src/ATokenCheckpointed.sol

```diff
@@ -0,0 +1,307 @@
+// SPDX-License-Identifier: UNLICENSED
```

Comment:
Took care of the compiler warnings in aa6ca4c

Reply:
Hmm I'm still seeing a few, and one error now too (error is from the `_transfer` suggestion, so ignore that for the sake of this convo and we'll move that to the other convo)

```
error[4334]: TypeError: Trying to override non-virtual function. Did you forget to add "virtual"?
   --> lib/aave-v3-core/contracts/protocol/tokenization/AToken.sol:206:3:
    |
206 |   function _transfer(
    |   ^ (Relevant source part starts here and spans across multiple lines).
Note: Overriding function is here:
   --> src/ATokenCheckpointed.sol:308:3:
    |
308 |   function _transfer(
    |   ^ (Relevant source part starts here and spans across multiple lines).



warning[2018]: Warning: Function state mutability can be restricted to view
  --> test/FractionalPool.t.sol:87:3:
   |
87 |   function _commonFuzzerAssumptions(address _address, uint256 _voteWeight, uint8 _supportType)
   |   ^ (Relevant source part starts here and spans across multiple lines).



warning[2018]: Warning: Function state mutability can be restricted to view
   --> test/GovernorCountingFractional.t.sol:172:3:
    |
172 |   function _randomSupportType(uint256 salt) public returns (uint8) {
    |   ^ (Relevant source part starts here and spans across multiple lines).



warning[2018]: Warning: Function state mutability can be restricted to view
   --> test/GovernorCountingFractional.t.sol:176:3:
    |
176 |   function _randomVoteSplit(FractionalVoteSplit memory _voteSplit)
    |   ^ (Relevant source part starts here and spans across multiple lines).

```

Reply:
Yeah, the error is expected. I changed the Aave code locally so that I could override `_transfer` and make the test I wrote pass. The compiler error will continue until we upstream the change and bump the aave version here.

Other warnings fixed in bbfaca1